File: tests/benchmarks/aot_c90/kernels/scalar_i64_loop.c
#include <stdint.h>
#include <stdio.h>

static int64_t run(int64_t n) {
    int64_t acc = 0;
    int64_t i = 0;
    while (i < n) {
        acc = acc + ((i * 31) % 1000003);
        i = i + 1;
    }
    return acc;
}

int main(void) {
    printf("%lld\n", (long long) run(1000000));
    return 0;
}

File: tests/benchmarks/aot_c90/kernels/scalar_i64_loop.rs
// Rust reference for scalar_i64_loop.xr (statement-level isomorphic, safe std-only).
// Built by run_aot_c90_benchmarks.py: rustc -C opt-level=3 -C lto=fat -C panic=abort

fn run(n: i64) -> i64 {
    var mut acc: i64 = 0;
    var mut i: i64 = 0;
    while i < n {
        acc += (i * 31) % 1000003;
        i += 1;
    }
    acc
}

fn main() {
    println!("{}", run(1000000));
}
